gapic-generator-java/src/main/java/com/google/api/generator/gapic/protoparser/Parser.java

```diff
@@ -175,7 +180,10 @@ public static GapicContext parse(CodeGeneratorRequest request) {
             mixinServices,
             transport);
 
-    Preconditions.checkState(!services.isEmpty(), "No services found to generate");
+    if (services.isEmpty()) {
+      LOGGER.warning("No services found to generate. This will produce an empty generated srcjar");
```

Comment:
I think the srcjar creation is part of `protoc`. I tried the following main function:

```java
  public static void main(String[] args) throws IOException {
    ExtensionRegistry registry = ExtensionRegistry.newInstance();
    ProtoRegistry.registerAllExtensions(registry);
    CodeGeneratorRequest request = CodeGeneratorRequest.parseFrom(System.in, registry);
    CodeGeneratorResponse response = Generator.generateGapic(request);
    if (response.getFileCount() > 0) {
      // we only write to stdout when we have files to generate. This is in order
      // to avoid creating an empty srcjar
      response.writeTo(System.out);
    }
  }
```

This still produces an empty srcjar. I'm looking into where do we parse `java_gapic_out`

Reply:
Correction - this is done in `Writer.write`. I'll adjust the logic.